src/pages/HomeSelector.vue

```diff
@@ -145,6 +146,16 @@ watch(
   { immediate: true },
 );
 
+watch(userClaims, (updatedUserClaims) => {
+  if (updatedUserClaims && updatedUserClaims.value) {
+    const { adminUid, assessmentUid, roarUid } = updatedUserClaims.value.claims;
+    const { userType } = useUserType(updatedUserClaims);
+    logProfileEvent('User claims updated', {
+      data: { adminUid, assessmentUid, roarUid, userType },
+    });
+  }
+});
```

Comment:
```suggestion
watch(userClaims, (updatedUserClaims) => {
  if (updatedUserClaims.value) {
    const { adminUid, assessmentUid, roarUid } = updatedUserClaims.value.claims;
    const { userType } = useUserType(updatedUserClaims);
    logProfileEvent('User claims updated', {
      data: { adminUid, assessmentUid, roarUid, userType },
    });
  }
});
```
I believe that we can just check for `.value` here. Assuming `userClaims` is a Vue `ref`, it will always be truthy (if it's not a `ref` then disregard, but it's being watched so I assume it is).

Someone please correct me if I am wrong.

Reply:
You've got a good point @ksmontville, but to be safe I would say to just use optional chaining

```diff
- if (updatedUserClaims && updatedUserClaims.value) {
+ if (updatedUserClaims?.value) {
```

Reply:
Same nit-pick as the other comment, as `logProfileEvent` is only used once, I would combine this into `logAuthEvent`